### ISSUE TITLE
Cleanup and improve library refresh.

### ISSUE THREAD

**Issue by YurySolovyov** [KeitIG/museeks#113]
Library refresh function is ~70 lines now and it is a single function.
It is very nested and uses mix of sync and async code.
We could do better by making it flatter and truly async using promises. That might also give us some speed boost since more of that would happen in parallel.

**Comment by KeitIG** [comment]
True. I've already done some nice improvements by splitting things, but I know this could be improved.

**Comment by KeitIG** [comment]
fixed by #123
